src/fxos/read_gaia_commit.js

```diff
@@ -0,0 +1,39 @@
+let adb = require('../adb');
+let assert = require('assert');
+let debug = console.log.bind(console, '[fxos/read_gaia_commit]');
+let denodeify = require('denodeify');
+let exec = require('mz/child_process').exec;
+let fs = require('mz/fs');
+let tmpdir = denodeify(require('tmp').dir);
+
+let roots = ['/data/local/', '/system/b2g/'];
+
+module.exports = async function readGaiaCommit() {
+  let local = await tmpdir();
+  debug('tmpdir', local);
+  await Promise.all(
```

Comment:
Instead of using `.all`, what if we just `.race` to the first one that can pull application.zip without error?